x-pack/plugins/apm/public/components/fleet_integration/apm_agents/runtime_attachment/supported_agents/java_runtime_attachment.tsx

```diff
@@ -0,0 +1,276 @@
+  '1.27.0',
+  '1.26.0',
+  '1.25.0',
+  '1.24.0',
+  '1.23.0',
+  '1.22.0',
+  '1.21.0',
+  '1.20.0',
+  '1.19.0',
+  '1.18.1',
+  '1.18.0',
+  '1.18.0.RC1',
+  '1.17.0',
+  '1.16.0',
+  '1.15.0',
+  '1.14.0',
+  '1.13.0',
+  '1.12.0',
+  '1.11.0',
+  '1.10.0',
+  '1.9.0',
+  '1.8.0',
+  '1.7.0',
+  '1.6.1',
+  '1.6.0',
+  '1.5.0',
+  '1.4.0',
+  '1.3.0',
+  '1.2.0',
+];
+
+function getApmVars(newPolicy: NewPackagePolicy) {
+  return newPolicy.inputs.find(({ type }) => type === 'apm')?.vars;
+}
+
+export function JavaRuntimeAttachment({ newPolicy, onChange }: Props) {
+  const [isDirty, setIsDirty] = useState(false);
+  const onChangePolicy = useCallback(
+    (runtimeAttachmentSettings: RuntimeAttachmentSettings) => {
+      const apmInputIdx = newPolicy.inputs.findIndex(
+        ({ type }) => type === 'apm'
+      );
+      onChange({
+        isValid: true,
+        updatedPolicy: {
+          ...newPolicy,
+          inputs: [
+            ...newPolicy.inputs.slice(0, apmInputIdx),
+            {
+              ...newPolicy.inputs[apmInputIdx],
+              vars: {
+                ...newPolicy.inputs[apmInputIdx].vars,
+                java_attacher_enabled: {
+                  value: runtimeAttachmentSettings.enabled,
+                  type: 'bool',
+                },
+                java_attacher_discovery_rules: {
+                  type: 'yaml',
+                  value: encodeDiscoveryRulesYaml(
+                    runtimeAttachmentSettings.discoveryRules
+                  ),
+                },
+                java_attacher_agent_version: {
+                  type: 'text',
+                  value: runtimeAttachmentSettings.version,
+                },
+              },
+            },
+            ...newPolicy.inputs.slice(apmInputIdx + 1),
+          ],
+        },
+      });
+      setIsDirty(true);
+    },
+    [newPolicy, onChange]
+  );
+
+  const apmVars = useMemo(() => getApmVars(newPolicy), [newPolicy]);
+
+  return (
+    <RuntimeAttachment
+      operationTypes={[
+        {
+          operation: {
+            value: 'include',
+            label: i18n.translate(
+              'xpack.apm.fleetIntegration.javaRuntime.operationType.include',
+              { defaultMessage: 'Include' }
+            ),
+          },
+          types: includeOptions,
+        },
+        {
+          operation: {
+            value: 'exclude',
+            label: i18n.translate(
+              'xpack.apm.fleetIntegration.javaRuntime.operationType.exclude',
+              { defaultMessage: 'Exclude' }
+            ),
+          },
+          types: excludeOptions,
+        },
+      ]}
+      onChange={onChangePolicy}
+      toggleDescription={i18n.translate(
+        'xpack.apm.fleetIntegration.javaRuntime.toggleDescription',
+        {
+          defaultMessage:
+            'Attach the Java agent to running and starting Java applications.',
+        }
+      )}
+      discoveryRulesDescription={
+        <FormattedMessage
```

Comment:
Can we use `i18n.translate` instead of `FormattedMessage`?

Reply:
not really, since I need to pass a react component as value and not a string.